tfhe/src/shortint/engine/server_side/add.rs

```diff
@@ -27,7 +27,7 @@ impl ShortintEngine {
     pub(crate) fn smart_add(
         &mut self,
         server_key: &ServerKey,
-        ct_left: &mut Ciphertext,
```

Comment:
Normally all the smart functions should take &mut for both input to clean carries if needed,
but these one do not seems to be correctly written

Reply:
Ok
Do you think we should fix this code by cleaning carries:
- in this PR (not sure how big this would be)?
- in another fix PR and in the meantime:
  - keep `mut` and silence this warning for the smart functions?
  - remove `mut` (as in this PR) and reintroduce it in the fix PR?

If we only clean carries of outputs, is it not enough (as all inputs will then be clean)? Or maybe we want to do it only when really needed to avoid it in some cases

Reply:
smart_ functions are meant to clean carries of input only when needed

I would say we keep the changes in the PR and try to find some time later to fix the shortint smart functions that
needs to be fixed